src/app/productConfig/productConfig.modal.js

```diff
@@ -49,13 +49,29 @@ class ModalInstanceCtrl {
 
     if ( this.selectableAmounts.indexOf( this.itemConfig.amount ) === -1 ) {
       this.customAmount = this.itemConfig.amount;
+      this.customInputActive = true;
     }
 
     if(!this.itemConfig['recurring-day-of-month'] && nextDrawDate) {
       this.itemConfig['recurring-day-of-month'] = Number(nextDrawDate.split('-')[2]).toString();
     }
   }
 
+  $onInit() {
+    this.handlePatternValidation = (() => {
```

Comment:
This seems like a little bit of a hack on `ng-pattern`. I guess it's fine as it's pretty clear what you are doing and I don't think this will break.

You could consider doing something like [this](https://github.com/CruGlobal/give-web/blob/master/src/common/components/paymentMethods/creditCardForm/creditCardForm.component.js#L73-L102) but it is a bit of code as you have to make sure the form is defined before adding something to the `$validators` array of the input. https://docs.angularjs.org/api/ng/type/ngModel.NgModelController#$validators

I'm fine with you leaving it though.